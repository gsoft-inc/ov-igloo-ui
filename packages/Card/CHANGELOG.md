# @igloo-ui/card

## 0.0.2

### Patch Changes

- fa1a096: ## Card component

  Cards are used a the background that contains elements of a parent ensemble.

  ### Component API

  | name     | description                                                                                |
  | -------- | ------------------------------------------------------------------------------------------ |
  | children | The content to display inside the component                                                |
  | size     | Changes the size of card, giving more or less padding `xsmall`, `small`, `medium`, `large` |
  | dataTest | Â Add a data-test tag for automated tests                                                   |
