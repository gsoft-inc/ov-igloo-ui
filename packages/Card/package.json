{
  "name": "@igloo-ui/card",
  "version": "0.2.0",
  "main": "dist/Card.js",
  "module": "dist/Card.js",
  "types": "dist/Card.d.ts",
  "styles": "dist/card.css",
  "publishConfig": {
    "access": "public"
  },
  "repository": "git+https://github.com/gsoft-inc/ov-igloo-ui.git",
  "author": "Officevibe Inc.",
  "contributors": [
    "Franck franck.gaudin@gsoft.com"
  ],
  "license": "Apache-2.0",
  "peerDependencies": {
    "react": ">= 16.8.6"
  },
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "rollup -c rollup.config.js"
  },
  "dependencies": {
    "@igloo-ui/tokens": "^2.0.0",
    "@hopper-ui/tokens": "^3.1.3",
    "classnames": "^2.3.2"
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "IE >= 11"
  ]
}
