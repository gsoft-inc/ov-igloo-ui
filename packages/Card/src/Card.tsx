import * as React from 'react';
import classNames from 'classnames';

import './card.scss';

export type Size = 'xsmall' | 'small' | 'medium' | 'large';
export type Ref = HTMLDivElement;

export interface CardProps extends React.ComponentPropsWithRef<'div'> {
  /** The content to display inside the card */
  children: React.ReactNode;
  /**  Changes the size of card, giving more or less padding */
  size?: Size;
  /** Add a data-test tag for automated tests */
  dataTest?: string;
}

const Card: React.FunctionComponent<CardProps> = React.forwardRef<
  Ref,
  CardProps
>(({ children, className, dataTest, size = 'large', ...rest }, ref) => {
  const classes = classNames('ids-card', className, {
    [`ids-card--${size}`]: size !== 'large',
  });

  return (
    <div ref={ref} className={classes} data-test={dataTest} {...rest}>
      {children}
    </div>
  );
});

export default Card;
