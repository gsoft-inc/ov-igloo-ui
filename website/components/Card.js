import Link from 'next/link';
import cx from 'classnames';

import Title from './Title';

export default function Card({
  inline,
  component,
  icon,
  title,
  description,
  link,
  className,
  featureFlag,
}) {
  const hasAction = typeof link === 'string' && !featureFlag;

  const card = (
    <div
      className={cx('io-card', className, {
        'io-card--inline': inline,
        'io-card--component': component,
      })}
    >
      <div className="io-card__img">{icon}</div>
      <div className="io-card__content">
        <Title as="h3" level={inline ? 4 : 3} className="io-card__title">
          {title}
        </Title>
        {description && <p className="io-card__description">{description}</p>}
      </div>
    </div>
  );
  return hasAction ? (
    <Link href={link}>
      <a className="io-card__link">{card}</a>
    </Link>
  ) : (
    card
  );
}
