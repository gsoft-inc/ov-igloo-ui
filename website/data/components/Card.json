[
  [
    {
      "tags": {},
      "filePath": "/Users/franck.gaudin/Devel/Igloo/components/packages/Card/src/Card.tsx",
      "description": "",
      "displayName": "Card",
      "methods": [],
      "props": {
        "children": {
          "defaultValue": null,
          "description": "The content to display inside the card",
          "name": "children",
          "parent": {
            "fileName": "components/packages/Card/src/Card.tsx",
            "name": "CardProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Card/src/Card.tsx",
              "name": "CardProps"
            },
            {
              "fileName": "components/node_modules/@types/react/index.d.ts",
              "name": "TypeLiteral"
            }
          ],
          "required": false,
          "type": { "name": "ReactNode" }
        },
        "size": {
          "defaultValue": null,
          "description": "Changes the size of card, giving more or less padding",
          "name": "size",
          "parent": {
            "fileName": "components/packages/Card/src/Card.tsx",
            "name": "CardProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Card/src/Card.tsx",
              "name": "CardProps"
            }
          ],
          "required": false,
          "type": { "name": "Size" }
        },
        "dataTest": {
          "defaultValue": null,
          "description": "Add a data-test tag for automated tests",
          "name": "dataTest",
          "parent": {
            "fileName": "components/packages/Card/src/Card.tsx",
            "name": "CardProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Card/src/Card.tsx",
              "name": "CardProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        }
      }
    }
  ]
]
