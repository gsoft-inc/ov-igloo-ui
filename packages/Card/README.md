# Card

Cards are used to group similar concepts and tasks to make it easier for users to scan, read, and get things done.

<Example />

<ReferenceLinks />

## Installation

To install `@igloo-ui/card` in your project, you will need to run the following command using [npm](https://www.npmjs.com/):

```bash
npm install @igloo-ui/card
```

If you prefer [Yarn](https://classic.yarnpkg.com/en/), use the following command instead:

```bash
yarn add @igloo-ui/card
```

## Usage

Then to use the component in your code just import it!

```jsx
import Card from '@igloo-ui/card';

<Card>Card content</Card>;
```
