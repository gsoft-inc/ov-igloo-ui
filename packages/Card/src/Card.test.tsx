/**
 * @jest-environment jsdom
 */
import React from 'react';
import { shallow } from 'enzyme';

import Card from './Card';

describe('Card', () => {
  const component = shallow(<Card>Hello world</Card>);
  test('It should render without errors', () => {
    const wrapper = component.find('.ids-card');
    expect(wrapper.length).toBe(1);
  });

  test('It should render a snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
