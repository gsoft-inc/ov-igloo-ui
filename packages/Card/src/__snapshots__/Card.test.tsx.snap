// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card It should render a snapshot 1`] = `
<DocumentFragment>
  <div
    class="ids-card"
  >
    Hello world
  </div>
</DocumentFragment>
`;
